### PR TITLE
.rustfmt.toml: brand new configuration to help with readability

### PR DESCRIPTION
**Pull request by unleashed**
I think this is probably the best time to start this given no PRs are pending.
    
In my tests this is probably the configuration that I felt produced the best results in readability without being perfect nor extreme.

### CODE REVIEW

**Comment by unleashed** [comment]
Appveyor is pulling a broken nightly, so don't worry about it.

File: .rustfmt.toml [hunk 1]
```diff
@@ -1,4 +1,15 @@
 version = "Two"
+edition = "2018"
 unstable_features = true
 error_on_line_overflow = true
 error_on_unformatted = true
+max_width = 110
```
Comment (davidor) [inline .rustfmt.toml]: Why 110?

File: examples/curl-report.rs [hunk 2]
```diff
@@ -15,55 +19,43 @@ use curl::easy::Easy;
...
+    let usages = [("metric11", 11),
+                  ("metric12", 12),
+                  ("metric21", 21),
+                  ("metric22", 22),
+                  ("metric31", 31),
+                  ("metric32", 32),
+                  ("metric41", 41),
+                  ("metric42", 42),
+                  ("metric51", 51),
+                  ("metric52", 52)].chunks(2)
```
Comment (davidor) [inline examples/curl-report.rs]: :scream_cat:

File: .rustfmt.toml [hunk 3]
```diff
@@ -1,4 +1,15 @@
 version = "Two"
+edition = "2018"
```
Comment (PhilipGough) [inline .rustfmt.toml]: What does this specify?

Comment (davidor) [reply .rustfmt.toml]: It refers to the Rust edition: https://doc.rust-lang.org/edition-guide/editions/index.html

File: .rustfmt.toml (hunk 1 above)
Comment (unleashed) [reply .rustfmt.toml]: same argument as above, it is reasonable, not too long or short, and made this codebase look the best in the experiments I ran.

**Comment by unleashed** [comment]
bors r=@davidor p=10

**Pull request merged**